spec/Section 5 -- Validation.md

```diff
@@ -249,19 +249,22 @@ query getName {
 
 **Formal Specification**
 
-* For each subscription operation definition {subscription} in the document
-* Let {subscriptionType} be the root Subscription type in {schema}.
-* Let {selectionSet} be the top level selection set on {subscription}.
-* Let {variableValues} be the empty set.
-* Let {groupedFieldSet} be the result of
-  {CollectFields(subscriptionType, selectionSet, variableValues)}.
-* {groupedFieldSet} must have exactly one entry, which must not be an
-  introspection field.
+* For each subscription operation definition {subscription} in the document:
+  * Let {subscriptionType} be the root Subscription type in {schema}.
+  * Let {selectionSet} be the top level selection set on {subscription}.
+  * Let {groupedFieldSet} be the result of
+    {CollectFields(subscriptionType, selectionSet, null)}.
```

Comment:
This might be scary, but I think the right thing to do would be to no longer rely on `CollectFields`, which is intended for runtime (hence the requirement to have variables)

You could probably get hand-wavy with the spec text and just say something about "recursively follow fragment spreads", but I think even if you were to copy and pair-down the `CollectFields` algorithm to only look for one field (encountering none with `@skip/@include` along the way) that it might not be too burdensome.

Reply:
It was [your suggestion to use `CollectFields()`](https://github.com/graphql/graphql-js/pull/2861#discussion_r614275790) that made me realise the issue with `@skip`/`@include` for subscriptions. I'll add a similar algorithm.